# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1465407639 25200
#      Wed Jun 08 10:40:39 2016 -0700
# Node ID 2049e123bf7b3536425e1970d41db911e399f40e
# Parent  f250714c629d36fbd3a1c9155935cc5bfeb07241
uigetdir, uigetfile, uiputfile: Return zeros when user cancels (bug #48171)

* octave-link.cc (__octave_link_file_dialog__): Return three zeros for Malab
compatibility when the file list starts with an empty string, meaning the
dialog was canceled.

diff --git a/libinterp/corefcn/octave-link.cc b/libinterp/corefcn/octave-link.cc
--- a/libinterp/corefcn/octave-link.cc
+++ b/libinterp/corefcn/octave-link.cc
@@ -226,27 +226,31 @@ Undocumented internal function.\n\
                                 multi_on);
 
   nel = items_lst.size ();
 
   // If 3, then retval is filename, directory, and selected index.
   if (nel <= 3)
     {
       int idx = 0;
-      for (std::list<std::string>::iterator it = items_lst.begin ();
-           it != items_lst.end (); it++)
+      if (items_lst.front ().empty ())
         {
-          retval(idx++) = *it;
+          retval(idx++) = 0;
+          retval(idx++) = 0;
+          retval(idx++) = 0;
+        }
+      else
+        for (std::list<std::string>::iterator it = items_lst.begin ();
+             it != items_lst.end (); it++)
+          {
+            retval(idx++) = *it;
 
-          if (idx == 1 && retval(0).string_value ().length () == 0)
-            retval(0) = 0;
-
-          if (idx == 3)
-            retval(2) = atoi (retval(2).string_value ().c_str ());
-        }
+            if (idx == 3)
+              retval(2) = atoi (retval(2).string_value ().c_str ());
+          }
     }
   else
     {
       // Multiple files.
       nel = items_lst.size () - 2;
       Cell items (dim_vector (1, nel));
 
       std::list<std::string>::iterator it = items_lst.begin ();
